### PR TITLE
Auto Retry

### PR DESCRIPTION
**Pull request by chris-rudmin**
Small refactor and added retry as well as more request options

### CODE REVIEW

**Comment by chris-rudmin** [comment]
@jpjoyal @spasiu @alavers @daudich

File: lib/attachments.js [hunk 1]
```diff
@@ -26,21 +12,27 @@ function Attachments (apiKey) {
  * Sample attachment object can be found in the docs:
  * http://developer.helpscout.net/help-desk-api/conversations/create-attachment/
  *
- * @param {Object} attachment
+ * @param {Object} options
+ *   @param {Object} attachment
+ *   @param {Number} timeout
```
Comment (dannytranlx) [inline lib/attachments.js]: Can you fix those indentations?

File: lib/attachments.js [hunk 2]
```diff
@@ -26,21 +12,27 @@ function Attachments (apiKey) {
...
+
+    this.helpscout.request({
+        method: 'post',
+        path: '/attachments.json',
+        data: options.attachment,
+        timeout: options.timeout,
+        callback: callback
     });
-};
+};
```
Comment (dannytranlx) [inline lib/attachments.js]: New line at end of file!

File: lib/conversations.js [hunk 3]
```diff
@@ -1,55 +1,41 @@
...
 }
 
 /**
  * List the mailbox's conversations.
  *
  * @param {Object} options
- *   @param {Number} page
+ *   @param {String} mailboxId
+ *   @param {Object} query
+ *     @param {Number} page
```
Comment (dannytranlx) [inline lib/conversations.js]: Indentation!

File: lib/index.js [hunk 4]
```diff
@@ -1,19 +1,109 @@
...
- * @return {Helpscout|Mailbox}
+ * @param {Object} config
+ *   @param {Number} maxRetries
+ *   @param {Number} defaultRetryDelay
+ *   @param {String} apiVersion
+ *   @param {String} apiRoot
+ *   @param {Object} query
+ *   @param {String} mailboxId
+ *   @param {String} apiKey
+ *   @param {Object} retryList
```
Comment (dannytranlx) [inline lib/index.js]: Indentation!

**Comment by dannytranlx** [comment]
Disclaimer, I haven't found anything wrong from my side, so I'll nitpick on codestyle :)
Great work! Really clean!

:+1:

**Pull request merged**